drivers/gpu/drm/i915/i915_active.c

```diff
@@ -130,6 +131,58 @@ static inline void debug_active_assert(struct i915_active *ref) { }
 
 #endif
 
+#if defined(__FreeBSD__)
+static struct llist_head linux_kmem_cache_list = LLIST_HEAD_INIT();
+
+struct cache_addr {
+	struct linux_kmem_cache *cache;
+	void *addr;
+};
+
+static void
+linux_kmem_cache_free_async_fn(struct irq_work *irqw)
+{
+	struct llist_node *freed_addr;
+
+	while ((freed_addr = llist_del_first(&linux_kmem_cache_list)) != NULL) {
+		struct cache_addr *freed_addr_ca = (struct cache_addr *) freed_addr;
+		kmem_cache_free(freed_addr_ca->cache, freed_addr_ca->addr);
+		kfree(freed_addr_ca);
+	}
+}
+static DEFINE_IRQ_WORK(linux_kmem_cache_free_async_work,
+	linux_kmem_cache_free_async_fn);
+
+static void
+linux_kmem_cache_free_async(struct linux_kmem_cache *c, void *m)
+{
+	struct cache_addr *ca;
+
+	if (m == NULL)
+		return;
+
+	ca = kmalloc(sizeof(struct cache_addr), GFP_KERNEL);
```

Comment:
It is not allowed to use malloc here. The trick is to reuse already allocated memory area pointed by m to store linked list metadata. This memory is intended to be freed, so we can use it before actual uma_zfree() is called.

Reply:
Thanks! I have incorporated this in my latest changes.